File: day7/Sort Colors/SortColors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        left = cur = 0
        right = len(nums) - 1
        while cur <= right:
            if nums[cur] == 0:
                nums[cur], nums[left] = nums[left], nums[cur]
                left += 1
                cur += 1
            elif nums[cur] == 1:
                cur += 1
            elif nums[cur] == 2:
                nums[cur], nums[right] = nums[right], nums[cur]
                right -= 1
        return

File: day7/Sort Colors/SortColors.java
class Solution {
    public void sortColors(int[] nums) {
        int cur = 0;
        int left = 0;
        int right = nums.length - 1;
        while (cur <= right) {
            if (nums[cur] == 0) {
                int tmp = nums[cur];
                nums[cur] = nums[left];
                nums[left] = tmp;
                left++;
                cur++;
            }
            else if (nums[cur] == 2) {
                int tmp = nums[cur];
                nums[cur] = nums[right];
                nums[right] = tmp;
                right--;
            }
            else {
                cur++;
            }
        }

    }
}
